shared/templates/mount_option_home/ansible.template

```diff
@@ -0,0 +1,64 @@
+# platform = multi_platform_all
+# reboot = false
+# strategy = configure
+# complexity = low
+# disruption = high
+
+-   name: "Initialize variables"
+    ansible.builtin.set_fact:
+        non_allowed_partitions: ["/", "/lib", "/opt", "/usr", "/bin", "/sbin", "/boot", "/dev", "/proc"]
+        home_directories: []
+        allowed_mount_point: []
+        fstab_mount_point_info: []
+
+-   name: "Get home directories from passwd"
+    ansible.builtin.getent:
+        database: passwd
+
+-   name: "Filter home directories based on UID range"
+    ansible.builtin.set_fact:
+        home_directories: "{{ home_directories + [item.data[4]] }}"
+    when:
+        - item.data[4] is defined
+        - item.data[2]|int >= {{{ uid_min }}}
+        - item.data[2]|int != {{{ nobody_uid }}}
+        - item.data[4] not in non_allowed_partitions
+    with_items: "{{ ansible_facts.getent_passwd | dict2items(key_name='user', value_name='data')}}"
+
+-   name: "Gather mount points"
+    ansible.builtin.setup:
+        filter: ansible_mounts
+
+-   name: "Ensure mount options for home directories"
+    block:
+
+    -   name: "Obtain mount point using df and shell"
+        ansible.builtin.shell: |
+            df {{ item }} | awk '/^\/dev/ {print $6}'
+        register: df_output
+        with_items: "{{ home_directories }}"
+
+    -   name: "Set mount point for each home directory"
+        ansible.builtin.set_fact:
+            allowed_mount_point: "{{ allowed_mount_point + [item.stdout_lines[0]] }}"
+        with_items: "{{ df_output.results }}"
+        when: item.stdout_lines[0] != ""
```

Comment:
If `df` doesn't return anything or produces an error then the stdout_lines is empty and doesn't have `0` element which causes the Playbook to fail and terminate.

Reply:
Thank you for the review, I updated this